src/zenml/steps/utils.py

```diff
@@ -411,6 +411,8 @@ def Do(
         if return_type is not None:
             if isinstance(return_type, Output):
                 # Resolve named (and multi-) outputs.
+                if sum(1 for dummy_var in return_type.items()) == 1:
```

Comment:
Three super small things:
- Usually `_` is the name indicating that the variable is not used
- Iterating over a dict directly (instead of `.keys()`, `.items()` or `.values()` is the same as iterating over the keys, so in this case I think removing the `.items()` looks cleaner
- I don't remember why but I don't see any reason why the `output.items()` should be a generator, so I'd suggest changing that to a dict and using `len()` here as Alexej suggested
-

Reply:
what about just `len(list(return_type))` ? that seems clearest to me

Reply:
@schustmi Got it for the first two. 

For the third comment, when I run our `quickstart.py` with the `len` implementation, I get the following error:

```
Traceback (most recent call last):
  File "examples/quickstart/quickstart.py", line 75, in <module>
    pipeline.run()
  File "/Users/strickvl/coding/zenml/repos/zenml/src/zenml/pipelines/base_pipeline.py", line 324, in run
    ret = self.stack.orchestrator.run(self, run_name=run_name)
  File "/Users/strickvl/coding/zenml/repos/zenml/src/zenml/orchestrators/local/local_orchestrator.py", line 54, in run
    runner.run(tfx_pipeline, run_name)
  File "/Users/strickvl/coding/zenml/repos/zenml/src/zenml/orchestrators/local/local_dag_runner.py", line 112, in run
    execute_step(component_launcher)
  File "/Users/strickvl/coding/zenml/repos/zenml/src/zenml/orchestrators/utils.py", line 67, in execute_step
    execution_info = tfx_launcher.launch()
  File "/Users/strickvl/.pyenv/versions/zenml-dev-3.8.6/lib/python3.8/site-packages/tfx/orchestration/portable/launcher.py", line 545, in launch
    executor_output = self._run_executor(execution_info)
  File "/Users/strickvl/.pyenv/versions/zenml-dev-3.8.6/lib/python3.8/site-packages/tfx/orchestration/portable/launcher.py", line 420, in _run_executor
    executor_output = self._executor_operator.run_executor(execution_info)
  File "/Users/strickvl/.pyenv/versions/zenml-dev-3.8.6/lib/python3.8/site-packages/tfx/orchestration/portable/python_executor_operator.py", line 133, in run_executor
    return run_with_executor(execution_info, executor)
  File "/Users/strickvl/.pyenv/versions/zenml-dev-3.8.6/lib/python3.8/site-packages/tfx/orchestration/portable/python_executor_operator.py", line 58, in run_with_executor
    result = executor.Do(execution_info.input_dict, output_dict,
  File "/Users/strickvl/coding/zenml/repos/zenml/src/zenml/steps/utils.py", line 415, in Do
    if len(return_type.items()) == 1:
TypeError: object of type 'generator' has no len()
```

That error is fixed with my implementation as above.

Reply:
Ah my bad `return_type` isn't actually a dict, it's a separate object and `.items()` is something that we implemented so we still need to use `return_type.items()`. For my third part I was referring to actually changing the code inside the `StepOutput` class, so that `StepOutput.items()` doesn't return a generator anymore but a dictionary.
I also liked James' suggestion of implementing `__len__` and potentially even some for of indexing on the `StepOutput` class